pandas/core/groupby/groupby.py

```diff
@@ -1665,6 +1668,7 @@ def nth(self, n, dropna=None):
 
         # old behaviour, but with all and any support for DataFrames.
         # modified in GH 7559 to have better perf
+        n = cast(int, n)
```

Comment:
I would really really avoid the need to do this, you can instead assign to a new variable that is the correct type

Reply:
No problem. So in your mind would cast only be suitable for expressions without assignment or should we avoid altogether?

Reply:
I would really avoid having to cast at all; this is just plain confusing as its not 'code' but for annotation purposes.

Reply:
I've refactored the branches to appease the type checker without cast but also arguably make the code more clear. Makes the diff a little larger than originally but I think for the better